test/sqllogictest/system_configuration.slt

```diff
@@ -0,0 +1,46 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# This file is derived from the logic test suite in CockroachDB. The
+# original file was retrieved on June 10, 2019 from:
+#
+# The original source code is subject to the terms of the Apache
+# 2.0 license, a copy of which can be found in the LICENSE file at the
+# root of this repository.
+
+mode cockroach
+
+query TT
+SELECT * FROM mz_system_configuration;
+----
+
+statement ok
+ALTER SYSTEM SET wal_level TO logical
+
+query TT
+SELECT * FROM mz_system_configuration;
+----
+wal_level   {"String":"logical"}
```

Comment:
The value type here is weird. It's dependent on how exactly serde_json chooses to derive its Serialize impl for an arbitrary Rust enum. This means we effectively can't write a parser for it and is only consumable by humans eye-balling and deciding if it feels correct. Can we use something more SQL-y instead? Maybe a JSON type and ensure the ast Value can successfully be converted to a JSON datum? Dunno.

Reply:
Yeah that is a good point, this is a bad choice for the value. The closest equivalent to this table I could find in postgres is `pg_settings` which displays all the settings that you'd be able to set from `ALTER SYSTEM SET` (plus some more). https://www.postgresql.org/docs/current/view-pg-settings.html

The type of the values for that view is `TEXT` and they display it in a human readable form.
```
joe=# SELECT name, setting, pg_typeof(setting) FROM pg_settings WHERE name = 'wal_level' OR name = 'max_connections';
      name       | setting | pg_typeof 
-----------------+---------+-----------
 max_connections | 5000    | text
 wal_level       | logical | text
(2 rows)
```

What if we made the value for `mz_system_configuration` be `TEXT` and uses the `AstDisplay` impl for `mz_sql_parser::ast::Value`? That way it would be both human readable and easily parsable.

Reply:
SGTM. It should also include the `vartype` column from https://www.postgresql.org/docs/current/view-pg-settings.html

Reply:
Just pushed these changes.